src/main/java/org/truffleruby/core/range/RangeNodes.java

```diff
@@ -147,17 +147,12 @@ private Object eachInternal(VirtualFrame frame, RubyRange range, RubyProc block)
         }
 
         @Specialization
-        protected Object eachObject(VirtualFrame frame, RubyLongRange range, NotProvided block) {
-            return eachInternal(frame, range, null);
-        }
-
-        @Specialization
-        protected Object each(VirtualFrame frame, RubyObjectRange range, NotProvided block) {
-            return eachInternal(frame, range, null);
+        protected Object eachObject(VirtualFrame frame, RubyLongRange range, Nil block) {
+            return eachInternal(frame, range, block);
         }
 
         @Specialization
-        protected Object each(VirtualFrame frame, RubyObjectRange range, RubyProc block) {
+        protected Object each(VirtualFrame frame, RubyObjectRange range, Object block) {
             return eachInternal(frame, range, block);
         }
```

Comment:
Looks like there is a missing RubyIntRange specialization with no block here.
I'll fix it, seems it was already the case before, and fix the order to make it clearer.